fedimint-server/src/net/peers.rs

```diff
@@ -78,6 +80,52 @@ struct PeerConnectionStateMachine<M> {
     state: PeerConnectionState<M>,
 }
 
+struct PeerStatusQuery {
+    response: oneshot::Sender<PeerStatus>,
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum PeerStatus {
+    Connected,
+    Disconnected,
+}
+
+type PeerStatusChannelSender = Sender<PeerStatusQuery>;
+type PeerStatusChannelReceiver = Receiver<PeerStatusQuery>;
+
+#[derive(Clone)]
+pub struct PeerStatusClient(HashMap<PeerId, PeerStatusChannelSender>);
+
+impl PeerStatusClient {
+    pub async fn query_all_status(&self) -> HashMap<PeerId, anyhow::Result<PeerStatus>> {
+        let mut results = HashMap::new();
+        for (peer_id, sender) in self.0.iter() {
```

Comment:
If I'm seeing it right, this should be possible to be

```
self.0.iter().map(...).collect()
```

but when I was typing it I've noticed that this way (and the way it's written currently), it would be done serially. We should probably send all the requests first, then collect all the responses to let stuff happen in parallel.

Reply:
Could be maybe `self.0.iter().map(...send...).collect().into_iter().map(...receive...).collect()`. :)

Reply:
Look now